From e68a9e3e08f63241bf8b32dd7c48a2d769fb062a Mon Sep 17 00:00:00 2001
From: clownshen <clownshen@google.com>
Date: Thu, 16 Aug 2018 17:11:30 +0800
Subject: [PATCH 18/70] DO NOT MERGE New carrier config
 enhanced_4g_lte_title_variant_int

Using KEY_ENHANCED_4G_LTE_TITLE_VARIANT_INT for customizing the title of
Enhanced 4G LTE Mode settings

Bug: 112676117
Test: Manual test, setting config to 0, 1 or -1 to change the index for
verifying different variant title of Enhanced 4G LTE Mode settings.

Change-Id: I6d52537ef75d57aa2b24240bb57fc1e115bca415
---
 res/values/strings.xml                           | 10 +++++++++-
 src/com/android/phone/MobileNetworkSettings.java | 16 ++++++++++------
 2 files changed, 19 insertions(+), 7 deletions(-)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index f728109e2..e91b79e1b 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -517,7 +517,15 @@
     <!-- Mobile network 4G title [CHAR LIMIT=30] -->
     <string name="enhanced_4g_lte_mode_title">Enhanced 4G LTE Mode</string>
     <!-- Carrier variant of Enhaced 4G LTE Mode title.  [CHAR LIMIT=50] -->
-    <string name="enhanced_4g_lte_mode_title_variant">Advanced Calling</string>
+    <string-array name="enhanced_4g_lte_mode_title_variant">
+        <!-- 0: Default -->
+        <item>@string/enhanced_4g_lte_mode_title</item>
+        <!-- 1: Verizon -->
+        <item>Advanced Calling</item>
+        <!-- 2: O2 UK -->
+        <item>4G Calling</item>
+    </string-array>
+
     <!-- Mobile network 4G summary [CHAR LIMIT=80] -->
     <string name="enhanced_4g_lte_mode_summary">Use LTE services to improve voice and other communications (recommended)</string>
 
diff --git a/src/com/android/phone/MobileNetworkSettings.java b/src/com/android/phone/MobileNetworkSettings.java
index 6a26f76b6..75e3bfc9b 100644
--- a/src/com/android/phone/MobileNetworkSettings.java
+++ b/src/com/android/phone/MobileNetworkSettings.java
@@ -1173,15 +1173,19 @@ public class MobileNetworkSettings extends Activity  {
              * but you do need to remember that this all needs to work when subscriptions
              * change dynamically such as when hot swapping sims.
              */
-            boolean useVariant4glteTitle = carrierConfig.getBoolean(
-                    CarrierConfigManager.KEY_ENHANCED_4G_LTE_TITLE_VARIANT_BOOL);
-            int enhanced4glteModeTitleId = useVariant4glteTitle ?
-                    R.string.enhanced_4g_lte_mode_title_variant :
-                    R.string.enhanced_4g_lte_mode_title;
+            int variant4glteTitleIndex = carrierConfig.getInt(
+                    CarrierConfigManager.KEY_ENHANCED_4G_LTE_TITLE_VARIANT_INT);
+            String[] variantTitles = getContext().getResources()
+                    .getStringArray(R.array.enhanced_4g_lte_mode_title_variant);
+            // Default index 0 indicates the default title string
+            CharSequence enhanced4glteModeTitle = variantTitles[0];
+            if (variant4glteTitleIndex >= 0 && variant4glteTitleIndex < variantTitles.length) {
+                enhanced4glteModeTitle = variantTitles[variant4glteTitleIndex];
+            }
 
             mButtonPreferredNetworkMode.setEnabled(hasActiveSubscriptions);
             mButtonEnabledNetworks.setEnabled(hasActiveSubscriptions);
-            mButton4glte.setTitle(enhanced4glteModeTitleId);
+            mButton4glte.setTitle(enhanced4glteModeTitle);
             mLteDataServicePref.setEnabled(hasActiveSubscriptions);
             Preference ps;
             ps = findPreference(BUTTON_CELL_BROADCAST_SETTINGS);
-- 
2.17.1

